pyeudiw/tools/utils.py

```diff
@@ -198,3 +201,75 @@ def satisfy_interface(o: object, interface: type) -> bool:
         if callable(getattr(interface, cls_attr)) and not callable(getattr(o, cls_attr)):
             return False
     return True
+
+
+_HttpcParams_T = NamedTuple('_HttpcParams_T', [('ssl', bool), ('timeout', int)])
+
+
+def cacheable_get_http_url(cache_ttl: int, url: str, httpc_params: dict, http_async: bool = True) -> requests.Response:
+    """
+    Make a cached http GET request.
+    The cache duration is UP TO cache_ttl. The actual duration is always
+    below that threshold.
+    The cache is realized with an lru_cache, which does not natively support a
+    time-based cache. To realize it, we exapand the call with a timestamp
+    rounded at the desired time.
+    For example, if the cache_ttl is 1 hour, and we make a request at 14:32,
+    the recorded timestamp will be 14:00 and the cache will be hit for all
+    subsequent requests (with the same parameters) until 14:59.
+    At 15:00, we reach a cache miss and a new value will be inserted.
+
+    The minimum supported time to live is 1 second.
+
+    When the response is not 200, the content of the cache is invalidated
+    in order to not pollute the cache whit 4xx and 5xx responses.
+
+    IMPORTANT: this function has limited support for httpc_params.
+    This is because python does not allow a lru_cache with a dictionary argument.
+    Currently, the only supported arguments are:
+        httpc_params.connection.ssl: bool
+        httpc_params.session.timeout: int
+    and they MUST be defined. When this is not the case, ValueError is raised.
+    """
+    ssl: bool | None = httpc_params.get("connection", {}).get("ssl", None)
+    timeout: int | None = httpc_params.get("session", {}).get("timeout", None)
+    if (ssl is None) or (timeout is None):
+        raise ValueError(f"invalid parameter {httpc_params=}: ['connection']['ssl'] and ['session'].['timeout'] MUST be defined")
+    curr_time_s = time.time_ns() // 1_000_000_000
+    if cache_ttl != 0:
+        ttl_timestamp = curr_time_s // cache_ttl
+    else:
+        ttl_timestamp = curr_time_s
+    httpc_p_tuple = _HttpcParams_T(ssl, timeout)
+    resp = _lru_cached_get_http_url(ttl_timestamp, url, httpc_p_tuple, http_async=http_async)
+
+    if resp.status_code != 200:
+        _lru_cached_get_http_url.cache_clear()
+    return resp
+
+
+@lru_cache(2048)
```

Comment:
```suggestion
@lru_cache(os.getenv("PYEUDIW_LRU_CACHE_TTL", 2048))
```

please add this do the docs